cristin-organization/src/main/java/no/unit/nva/cristin/organization/DefaultOrgQueryClientProvider.java

```diff
@@ -0,0 +1,42 @@
+package no.unit.nva.cristin.organization;
+
+import no.unit.nva.model.Organization;
+import nva.commons.apigateway.RequestInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultOrgQueryClientProvider implements IClientProvider<RequestInfo, IQueryApiClient<Organization>> {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultOrgQueryClientProvider.class);
+
+    public static final String VERSION = "version";
+    public static final String VERSION_ONE = "1";
+    public static final String VERSION_TWO = "2";
+    public static final String CLIENT_WANTS_VERSION_OF_THE_API_CLIENT = "Client wants version {} of the api client";
+
+    @Override
+    public IQueryApiClient<Organization> getClient(RequestInfo requestInfo) {
+        if (clientRequestsVersionTwo(requestInfo)) {
+            logger.info(CLIENT_WANTS_VERSION_OF_THE_API_CLIENT, VERSION_TWO);
+            return getVersionTwo();
+        } else {
+            logger.info(CLIENT_WANTS_VERSION_OF_THE_API_CLIENT, VERSION_ONE);
+            return getVersionOne();
+        }
+    }
+
+    private boolean clientRequestsVersionTwo(RequestInfo requestInfo) {
+        return requestInfo
+                   .getQueryParameterOpt(VERSION)
```

Comment:
Do we want to control versions through query params, or rather use Accept header? E.g. `Accept: application/v2+json`

Reply:
I can change to using accept header, but is your example a valid example or do we need to discuss how we should format the accept header?